### ISSUE TITLE
Add alternatives to play a video from bytes.

### ISSUE THREAD

**Issue by CharlieFuu69** [renpy/renpy#5744]
Why did I write that issue title? One of the many things I'm using in Ren'Py is its ability to play in-game videos, and I've noticed a very peculiar situation that occurs when I run the Ren'Py game on an HDD storage unit. The video runs normally, but every so often, the disk enters a sort of "sleep phase", causing it to stop. When this happens, after a few seconds the video freezes until the disk starts spinning again (around 1 to 1.5 seconds). When it returns to the active phase, the video continues to play normally.

This situation does not occur in an SSD storage unit, probably attributed to the fact that a solid state drive has no mechanical parts, has better response times and is much faster in transfer speed than an HDD.
Now, if doubts arise about the type of video I use, it's not really much of a surprise because I've always been faithful to WebM when it comes to Ren'Py.

General details of the videos used:
```
- Container: WebM.
- Codec: VP8.
- Video size range: 50-120 MB.
- Bitrate: 4000-8000 kbps.
```

Now the situation is this. To get audio data from a byte array, we have `AudioData()`, while to get image data from a byte array, we have `im.Data()`. The relationship between this and the problem at hand is that I could use `renpy.file("filename.webm").read()` to save the bytes in a variable, and thus "preload" the video to minimize the problem noted.

The question is as follows: Could there be an alternative in the future to load videos from a byte array, in a similar way as with `AudioData()` and `im.Data()`?

Clearly the uses of `AudioData()` and `im.Data()` are for different things than a video, but I think you get the idea.

**Comment by renpytom** [comment]
Have you tried supplying AudioData to a video function? I haven't tried it, but I believe it will work.

**Comment by CharlieFuu69** [comment]
Suggest that loads the video of the video in `Audiodata()`, and in turn, use this `Audiodata()` in a class like `Movie()`?

It sounds like what I need, but it would be necessary to find out if this could work. When I do that test, I will inform you if it works ...

**Comment by CharlieFuu69** [comment]
UPDATE:

Mr. Pytom, it seems that this works... wonderful...
![image](https://github.com/user-attachments/assets/d241fb89-a1e0-4135-a660-66feedd18f26)

I have used the following code as a test:
```renpy
image MV_00_006 = Movie(play=AudioData(renpy.file("MV/MV_00_006.webm").read(), "MV/MV_00_006.webm"),
                                             loop=False,
                                             channel="movie_dp")
```

There is a small detail. Apparently, `Audiodata()` does not admit the use of partial playback, at least, not using the following format:
```python
## Partial playback not working
AudioData(b".........", "<from 2.1>MV_00_006.webm")
```

Although partial playback does not seem to have an effect on the `filename` parameter of `Audiodata()`, video can be taken effectively from an `Audiodata()` (bytes), which is a great step with the main problem of the issue. The partial playback of the video, only has an effect if `Movie()` takes as an argument a string with the video path:
```renpy
## Partial playback working
image MV_00_006 = Movie(play="<from 2.1>MV/MV_00_006.webm",
                                             loop=False,
                                             channel="movie_dp")
```

**Comment by renpytom** [comment]
I'll look into seeing if I can fix that.

**Comment by CharlieFuu69** [comment]
I think you could even modify the documentation, pointing out that `AudioData()` also supports loading videos from a byte string. At least for my part, I didn't think it would work, but it did =D

On the other hand, there is another no less important issue to deal with: the memory usage of the preloaded video. I would need to define some approach that allows to free the memory used by the loaded bytes of that video (`renpy.file()` in variable), at the time it ends.

**Comment by Andykl** [comment]
https://github.com/renpy/renpy/commit/003ac717d8e2941a148970ac3f95f0d8ce9dcb11 should fix that, you can test it in nightly.

**Comment by CharlieFuu69** [comment]
Great! The feature was finally implemented in `v8.3.0`.

I will close the current Issue as it has been fixed.
